### ISSUE TITLE
Upgrade to 0.2.0 no events in Elastic

### ISSUE THREAD

**Issue by kkellner** [legrego/homeassistant-elasticsearch#89]
After upgrading from 0.1.3 to 0.2.0 I needed to update the configuration as follows:

My version 0.1.3 config:

```yaml
elastic:
    url: SOME_URL
    publish_frequency: 2
    only_publish_changed: true
    rollover_max_age: 1d
    exclude:
      entities: [ 'sensor.es_publish_queue' ]
      domains: [ 'sun' ]
```
My version 0.2.0 config:
```yaml
elastic:
    # URL should point to your Elasticsearch cluster
    url: SOME_URL
    publish_frequency: 2
    only_publish_changed: true
    ilm_max_size: 10gb
    ilm_delete_after: 30d
    exclude:
      entities: [ 'sensor.es_publish_queue' ]
      domains: [ 'sun' ]
```

Home Assistant restarted without issues with the above change, however I am not getting any events (visible) in Kibana (Elasticsearch).   Is there an upgrade step I missed?   Here are my current indexes:

![image](https://user-images.githubusercontent.com/9540241/76819857-0ea7f180-67cf-11ea-9b9a-17793e435d7c.png)

Events stopped after 21:18:

![image](https://user-images.githubusercontent.com/9540241/76820200-16b46100-67d0-11ea-8745-7fd58ede0b92.png)

Does something need to be updated on the Elasticsearch side?

**Comment by kkellner** [comment]
Extra info:  This is what I get in the logic on the HA side:

```
Error publishing documents to Elasticsearch: ('3 document(s) failed to index.', [{'index': {'_index': 'hass-events-v4-000001', '_type': 'doc', '_id': '6bS56XAB1v_E5Tv-mP46', 'status': 400, 'error': {'type': 'mapper_parsing_exception', 'reason': 'Could not dynamically add mapping for field [hass.object_id.lower]. Existing mapping for [hass.object_id] must be of type object but found [keyword].'}, 'data': {'hass.domain': 'sensor', 'hass.object_id': 'apc_ups_output_voltage', 'hass.object_id.lower': 'apc_ups_output_voltage', 'hass.entity_id': 'sensor.apc_ups_output_voltage', 'hass.entity_id.lower': 'sensor.apc_ups_output_voltage', 'hass.attributes': {'state': 'Online', 'unit_of_measurement': 'V', 'friendly_name': 'APC_UPS Output Voltage', 'icon': 'mdi:flash'}, 'hass.value': 123.0, '@timestamp': datetime.datetime(2020, 3, 17, 18, 20, 19, 11143, tzinfo=<UTC>), 'agent.name': 'My Home Assistant', 'agent.type': 'hass', 'agent.version': '0.106.6', 'ecs.version': '1.0.0', 'host.geo.location': {'lat': xxxxx, 'lon': xxxxx}, 'host.architecture': 'x86_64', 'host.os.name': 'Linux', 'host.hostname': 'homeassistant', 'tags': ['hass']}}}, {'index': {'_index': 'hass-events-v4-000001', '_type': 'doc', '_id': '6rS56XAB1v_E5Tv-mP46', 'status': 400, 'error': {'type': 'mapper_parsing_exception', 'reason': 'Could not dynamically add mapping for field [hass.object_id.lower]. Existing mapping for [hass.object_id] must be of type object but found [keyword].'}, 'data': {'hass.domain': 'sensor', 'hass.object_id': 'apc_ups_load', 'hass.object_id.lower': 'apc_ups_load', 'hass.entity_id': 'sensor.apc_ups_load', 'hass.entity_id.lower': 'sensor.apc_ups_load', 'hass.attributes': {'state': 'Online', 'unit_of_measurement': '%', 'friendly_name': 'APC_UPS Load', 'icon': 'mdi:gauge'}, 'hass.value': 26.0, '@timestamp': datetime.datetime(2020, 3, 17, 18, 20, 19, 20310, tzinfo=<UTC>), 'agent.name': 'My Home Assistant', 'agent.type': 'hass', 'agent.version': '0.106.6', 'ecs.version': '1.0.0', 'host.geo.location': {'lat': 39.59101487054839, 'lon': -105.0809357687831}, 'host.architecture': 'x86_64', 'host.os.name': 'Linux', 'host.hostname': 'homeassistant', 'tags': ['hass']}}}, {'index': {'_index': 'hass-events-v4-000001', '_type': 'doc', '_id': '67S56XAB1v_E5Tv-mP46', 'status': 400, 'error': {'type': 'mapper_parsing_exception', 'reason': 'Could not dynamically add mapping for field [hass.object_id.lower]. Existing mapping for [hass.object_id] must be of type object but found [keyword].'}, 'data': {'hass.domain': 'sensor', 'hass.object_id': 'apc_ups_input_voltage', 'hass.object_id.lower': 'apc_ups_input_voltage', 'hass.entity_id': 'sensor.apc_ups_input_voltage', 'hass.entity_id.lower': 'sensor.apc_ups_input_voltage', 'hass.attributes': {'state': 'Online', 'unit_of_measurement': 'V', 'friendly_name': 'APC_UPS Input Voltage', 'icon': 'mdi:flash'}, 'hass.value': 123.0, '@timestamp': datetime.datetime(2020, 3, 17, 18, 20, 19, 31435, tzinfo=<UTC>), 'agent.name': 'My Home Assistant', 'agent.type': 'hass', 'agent.version': '0.106.6', 'ecs.version': '1.0.0', 'host.geo.location': {'lat': 39.59101487054839, 'lon': -105.0809357687831}, 'host.architecture': 'x86_64', 'host.os.name': 'Linux', 'host.hostname': 'homeassistant', 'tags': ['hass']}}}])
12:20:19 PM – custom_components/elastic/__init__.py (ERROR)
```

**Comment by kkellner** [comment]
Could it be related to this change by @legrego ?  I'm not well-versed in Elasticsearch so I'm just poking around.

https://github.com/legrego/homeassistant-elasticsearch/blame/890deabd60c119b374a2c11793b58906a0da7312/custom_components/elastic/__init__.py#L545

**Comment by ThomDietrich** [comment]
Hey all,
I'm seeing the same error on a fresh installation. This must be a newly introduced bug.

I'm using a fresh ELK stack via https://github.com/deviantony/docker-elk
I can see the index `active-hass-index-v4` being created but never filled.

The Home Assistant log shows the following error:

```log
2020-03-21 23:21:39 INFO (MainThread) [custom_components.elastic] Publishing 177 documents to Elasticsearch
2020-03-21 23:21:41 INFO (MainThread) [elasticsearch] POST http://xxxxxxxxxxxx.com:9200/_bulk [status:200 request:0.978s]
2020-03-21 23:21:41 ERROR (MainThread) [custom_components.elastic] Error publishing documents to Elasticsearch: ('177 document(s) failed to index.', [{'index': {'_index': 'active-hass-index-v4', '_type': '_doc', '_id': 'AEQv_3ABLXAWBJmS_sUb', 'status': 400, 'error': {'type': 'illegal_argument_exception', 'reason': "Can't merge a non object mapping [hass.entity_id] with an object mapping [hass.entity_id]"}, 'data': {'hass.domain': .........
```

**Comment by legrego** [comment]
Hey all,

Sorry for the delayed response, and that you’re having issues with the latest version. I’ll work on reproducing this and getting a fix out as time permits

**Comment by legrego** [comment]
I think I see the issue, but I have no clue how that made it into the release. I could have sworn I fixed this. The index mapping is using “fields” instead of “properties” for the entity_id and object_id fields. I’m on mobile now so I can’t verify, but if I’m right it’ll be an easy fix for me to push

**Comment by legrego** [comment]
I've published `v0.2.1` to address this. Before upgrading, ensure that you've remove any `hass-events-v4*` indices and templates, as they were created with invalid settings.
